mkosi

```diff
@@ -2830,25 +2865,46 @@ def check_root():
     if os.getuid() != 0:
         die("Must be invoked as root.")
 
+def run_shell(args):
+    target = "--directory=" + args.output if args.output_format in (OutputFormat.directory, OutputFormat.subvolume) else "--image=" + args.output
+
+    cmdline = ["systemd-nspawn",
```

Comment:
> Regarding --register=no: This is an interactive boot, and I think it's a really good thing that we can connetc from the outside to it. It's at least the way I usually work here: do something on the console of the container, and then get another shell in it via "machinectl shell" so that i can invoke strace or gdb or whatever else to supervise what I am doing there. By doing --register=no we'd take that away.

Hmm. A problem is that this makes it impossible to run this command simultaneously on different images in the default output path, since you can't have two machines with the same name `image`

(https://github.com/systemd/systemd/issues/6347 would prevent this even with `--register=no` but we can fix that one in systemd)